### DISCUSSION TITLE
Show resources in the FE instead of tweets

### DISCUSSION THREAD

**Discussion by Utkarshbhimte** [covidarmy/backend#34] [Ideas]
People are looking for resources, their old unverified tweets are not much help to them, we should move towards building an architecture that 

* can be linked to any input channel like Twitter or some can share a story and we can extract data via OCR there. 
* parses data from it, 
* verifies the resource by contacting them (either via bot or volunteers)
* ranks the resources according to their status in FE
* Follows up with the submitted resources for updates

This could be the first step to work towards a much better app

**Comment by SwarnimWalavalkar** [comment]
A very interesting avenue to explore... My thoughts on a couple of things you mentioned,

## Multiple Data Sources
We'll have to find data sources that are as real-time and as immediately useful/actionable as tweets.

### WApp/Insta Story OCR
One thing you suggested that could be a great addition is allowing people to share an Instagram or WhatsApp story, widespread distribution of stories is limited since those channels were primarily made for communication within small groups. Allowing people to upload it to our platform will enable a far more indexable and shareable channel... We could then also plug this into our verification infrastructure for added authenticity

#### Implementation
Here's one way to implement that,
- Upload an image to a Google Cloud Storage Bucket
- That triggers a Cloud Function that uses the Cloud Vision API to run OCR on the image
- The results are then stored in a MongoDB Database

### Vaccine Availability Notifications
Allow the user to opt-in to notifications for vaccine availability in their chosen city/district

#### Implementation
Use the [COWIN REST APIs](https://apisetu.gov.in/public/api/cowin#/Appointment%20Availability%20APIs/findByDistrict) to check slot availability, notify the user via text/email or push notifications on Android (using PWAs)

## Verification System
I think volunteers is the way to go here, we could set up a dashboard where volunteers can see all active leads, verify them and mark them as working, not-working, busy, invalid etc, Busy leads can be shown again after a cooldown period which increases with each subsequent busy mark until it is finally deemed invalid...

The volunteer recruitment process should be as systematic and automated as possible while also avoiding bots and spammers

**Reply by Utkarshbhimte** [reply]
We can sure discuss the implementation, later on, Let's decide if we want to go down this route or not. 

As then our focus would be to help volunteers with a better tool and also then our app would be dependent on volunteers as well. I don't think we can do this purely with tech. We don't have any feedback loop here. 

We should go towards building a MVP in that direction and then try to optimize it.

**Reply by rgxai** [reply]
Yes but we could still take this approach and parse the tweets to show the contact number in a more easy to access manner. We can show the tweet text and below that we can show some parsed fields and prompts (such as contact number, button to send a whatsapp to them, and prompts to take feedback on the lead status). Thoughts? Eventually we can have non tweet objects (manually added by volunteers, from OCR, etc) and remove the tweet text display all together.

**Comment by Utkarshbhimte** [comment]
@rgxai If we are verifying the leads then we don't need to show original tweets. it does not provide any value to the users.

So we have to make a "resources" collection which will have all the parse data and references on the source, but nothing much.

**Comment by Utkarshbhimte** [comment]
and also we can decide on the approach, and the interim milestones in a separate discussion, the objective is here to finalize if we go into that route or not.

Volunteer driven with verified resources.
We need to ask the volunteers who are doing it already and check if this is something that can help them and then try to support them in their website/app.

It would be hard to centralize such a human-driven mechanism in India, We should help local volunteers to set up and manage everything easily.

This includes a resource where we share how to use no-code tools to set up everything and how to link our API to their no-code tool as well

Where we will also provide WA bot features as services, get them leads from Twitter/Instagram/Forms to verify (local people would be best to have these conversations).

**Reply by rgxai** [reply]
Yes that's the eventual plan and I do think there's a lot of merit in moving towards this approach. The reason however we still would like to display the text from the original tweets is because we won't be able to easily parse data of the name from the tweets. For example, name of the supplier or the hospital. If you think that field is not important, we can discard the tweets and just show the parsed data too.

**Reply by Utkarshbhimte** [reply]
The validation team should take care of this data entry, right?

**Reply by rgxai** [reply]
So we don't show the lead till it is verified at all? Are we assuming that the verified tag put on the tweet would have no verification merit? And that tweet goes into the volunteer verification flow before showing up on our website / chatbot? that could work too in my opinion.

**Reply by Utkarshbhimte** [reply]
> So we don't show the lead till it is verified at all?
 
People want verified sources and they would go to Twitter directly when the verified does not work for them. Our Northstar metric is to show verified resources, and if needed in the future, we can have a separate tab for raw feed. Where people can mark the resources themselves as we mentioned in https://github.com/covidarmy/backend/issues/36.

> Are we assuming that the verified tag put on the tweet would have no verification merit? 

We can bump these up for the volunteers to verify themselves but I don't think it should get more attention on the user side

> And that tweet goes into the volunteer verification flow before showing up on our website / chatbot?

Yeah, and they can add their own review on the resource but if the cooldown gets decided on the volunteer's decision.

**Reply by rgxai** [reply]
Got it. Can you finalise your thoughts and flow suggestions for this so I can take this up into development ASAP?

**Reply by Utkarshbhimte** [reply]
Sure, I will try to get it done asap and share with the team
